src/Modules/Public/Public.service.ts

```diff
@@ -0,0 +1,132 @@
+import { CoinGateClient } from '#Modules/Client/CoinGate.client';
+
+import {
+  CurrencyKindEnum,
+  GetCurrenciesData,
+  GetExchangeRateData
+} from './types';
+
+/**
+ * Class representing a Public Service
+ * @extends CoinGateClient
+ */
+export class PublicService extends CoinGateClient {
+  /**
+   * Current exchange rate for any two currencies, fiat or crypto.
+   * @param {Object} params two currencies which exchange rate you want to get. Example: { from: 'GBP', to: 'EUR' }
+   * @returns exchange rate
+   */
+  public getExchangeRate(params: GetExchangeRateData) {
+    const path = this.buildPath({
+      path: '/v2/rates/merchant/:from/:to',
+      params
+    });
+
+    return this.get({ path });
+  }
+
+  /**
+   * Current CoinGate exchange rates for Merchants and Traders.
+   * @returns exchange rates
+   */
+  public listExchangeRates() {
+    return this.get({ path: '/v2/rates/' });
+  }
+
+  /**
+   * A health check endpoint for CoinGate API.
+   * @returns pong!
+   */
+  public ping() {
+    return this.get({ path: '/v2/ping/' });
+  }
+
+  /**
+   * Get IP addresses of CoinGate servers
+   * @param {string} separator
+   * @returns ip addresses
+   */
+  public ipAddresses(separator?: string) {
+    return this.get({ path: '/v2/ips-v4/', params: { separator } });
+  }
+
+  /**
+   * Get Currencies
+   * @param {Object} params
+   * @param {boolean} params.native
+   * @param {boolean} params.enabled
+   * @param {boolean} params.merchant_pay
+   * @param {boolean} params.merchant_receive
+   * @param {CurrencyKindEnum} params.kind
+   * @returns currencies
+   */
+  public getCurrencies(params?: GetCurrenciesData) {
+    return this.get({
+      path: '/v2/currencies/',
+      params
+    });
+  }
+
+  /**
+   * @returns checkout currencies
+   */
+  public getCheckoutCurrencies() {
+    return this.getCurrencies({
+      kind: CurrencyKindEnum.CRYPTO,
+      native: true,
+      merchant_pay: true
+    });
+  }
+
+  /**
+   * @param {CurrencyKindEnum} kind
+   * @returns merchant pay currencies
+   */
+  public getMerchantPayCurrencies(kind?: CurrencyKindEnum) {
+    return this.getCurrencies({
+      kind,
+      native: false,
+      merchant_pay: true
+    });
+  }
+
+  /**
+   * @param {CurrencyKindEnum} kind
+   * @returns merchant payout currencies
+   */
+  public getMerchantPayoutCurrencies(kind?: CurrencyKindEnum) {
```

Comment:
`currencies?&enabled=true&merchant_receive=true&kind=kind`
document in the readme that kind can be passed